### PR TITLE
Start Bastion changes

### LINKED ISSUE #41: Revert Bastion to OW1

### PR DESCRIPTION
**Pull request by snappycreeper**
Added sentry form with 300 ammo

Resolves #41

### CODE REVIEW

**Comment by MaxwellJung** [comment]
I'll review this in the evening when I get home from work.

In the meantime, can you please describe how your code works? See #24 for reference.

**Comment by snappycreeper** [comment]
"Bastion - Technical Tracker" - Tracks if bastion gets hacked, no more sentry mode GUI
"Bastion - Continuous Sentry Mode" - Bascailly, once the bastion enters sentry mode they will be sentry mode until the time runs out, which will auto cancel the animation and put him back in sentry mode. Once he runs out of the 300 ammo, the bastion has to press shift to exit the sentry mode which gives them a 10 second cooldown, and reloads their ammo.
"Bastion - Start Chasing Sentry Ammo" - Ammo tracking
"Bastion - Stop Chasing Sentry Ammo" - Ammo tracking
"Bastion - Progress Bar" - Displays the ammo GUI for the bastion, with a progress bar
"Bastion - Recon Mode" - Delete the GUI if the bastion goes into normal recon mode
"Bastion - Reload" - Reloads the bastions sentry mode
"Bastion - Turret Form Movement Speed" - Makes the bastion stay still in turret/sentry form
"Bastion - Effects Remove Sentry Mode" - Stuns, hacked, etc
 "Bastion - Recon Form Movement Speed" - Bastion normal speed if they aren't in turret form, so they won't get stuck

This should explain all of the rules well enough, once you get ingame and test it out you'll see what I mean. I also have a tank mode that I can add if that is something you want.

**Comment by MaxwellJung** [comment]
If you can develop the tank mode for bastion, that would be awesome!

**Comment by MaxwellJung** [comment]
For the HUD, I strongly recommend changing the visibility instead of creating/destroying the HUD. See the zarya.opy for example on HUD visibility.

File: src/heroes/bastion.opy [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+rule "Bastion - Continuous Sentry Mode":
+    @Event eachPlayer
+    @Hero bastion
+    @Condition eventPlayer.isUsingAbility1() == true
+    @Condition eventPlayer.BastionTankMode == false
+    @Condition eventPlayer.isDummy() == false
+    
+  
+    waitUntil(eventPlayer.isInAlternateForm() == false, 7.016)
```
Comment (MaxwellJung) [inline src/heroes/bastion.opy:25]: Where is this 7.016 from? Can you replace this with a keyword and define the value of the keyword in either "ow1_constants.opy' or "ow2_constants.opy" (aka avoid magic numbers)?

Comment (snappycreeper) [reply src/heroes/bastion.opy:25]: That is how long bastion's overwatch 2 turret form is

Comment (snappycreeper) [reply src/heroes/bastion.opy:25]: But yes, I will do that

**Comment by MaxwellJung** [comment]
Found another bug related to tank mode ult.

When you jump or fall from ledge while transitioning to ult, you can't move while in tank mode.

**Comment by snappycreeper** [comment]
Okay, ill look into that right now

**Comment by MaxwellJung** [comment]
nvm that bug's due to bad merge commit

**Pull request merged**